src/content/components/BugList/BugList.js

```diff
@@ -80,41 +74,64 @@ const priorityOptions = [
   },
 ];
 
-export class BugList extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      selectedBugs: {},
-      showResolved: props.showResolved,
-      editSelection: -1,
-      newIteration: currentIteration,
-      newPriority: "---",
+export const BugList = ({ ...props }) => {
+  const context = useContext(GlobalContext);
+  const [selectedBugs, setSelectedBugs] = useState({});
+  const [showResolved, setShowResolved] = useState(props.showResolved);
+  const [editSelection, setEditSelection] = useState(-1);
+  const [bugUpdating, setBugUpdating] = useState(false);
+
+  const mounted = useRef();
+  const bugsRef = useRef(props.bugs);
+  const iterationSelectRef = useRef(null);
+  const iterationRef = useRef(context.iterations.getIteration().number);
+  const priorityRef = useRef("---");
+
+  useEffect(() => {
+    if (!mounted.current) {
+      mounted.current = true;
+    } else if (
+      bugUpdating &&
+      JSON.stringify(props.bugs) != JSON.stringify(bugsRef.current)
+    ) {
+      setBugUpdating(false);
+      setEditSelection(-1);
+      setSelectedBugs({});
+    }
+
+    return () => {
+      bugsRef.current = props.bugs;
     };
-    this.onCheck = this.onCheck.bind(this);
-    this.onAllSelectedCheck = this.onAllSelectedCheck.bind(this);
-    this.onCheckShowResolved = this.onCheckShowResolved.bind(this);
-  }
+  }, [bugUpdating, props.bugs]);
+
+  const handleMenuOpen = useCallback(() => {
+    let { current } = iterationSelectRef;
+
+    requestAnimationFrame(() => {
+      current.focusedOptionRef?.scrollIntoView({ block: "nearest" });
+    });
+  }, []);
 
-  getRowClassName(bug) {
+  const getRowClassName = bug => {
```

Comment:
same here, keep the traditional `function () {}` syntax unless it's a function component (i.e. it returns JSX)

Reply:
also, this should be moved out of BugList. the way you've changed this, you're redefining all of these functions one very render. the content of a functional component is executed on every render. that's why you'd use `useCallback`. but for these kinds of things, it's better to define it outside of the component so React doesn't need to read over it every time. just change the parameters or something.

really, changing this to a functional component is not such a great idea to do in this PR. it's already becoming quite unwieldy to review. you should try to keep changes as minimal as possible, just as a general rule. I think we'd be better off making these general refactorings in a separate PR.